batteries/dq_dv.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jan  5 10:10:36 2023
+
+@author: mpopeil
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import sklearn as sk
+import sklearn.datasets
+from pandas import DataFrame
+from scipy.interpolate import Akima1DInterpolator
+import scipy
+
+from scipy import signal
+from scipy.signal import savgol_filter
+from scipy import interpolate
+from scipy.integrate import simpson
+from numpy import trapz
+
+#load data into dataframes
+files=['L2_Ta.csv']
+for data in files:
+    
+    df = pd.read_csv(data)
+# charge1=df[(df['Cycle']==1) & (df['Step']==4)]
+# discharge1=df[(df['Cycle']==1) & (df['Step']==7)]
+# charge2=df[(df['Cycle']==2) & (df['Step']==4)]
+# discharge2=df[(df['Cycle']==2) & (df['Step']==7)]
+# charge3=df[(df['Cycle']==3) & (df['Step']==4)]
+# discharge3=df[(df['Cycle']==3) & (df['Step']==7)]
+
+# cycles=[charge1,discharge1, charge2, discharge2, charge3, discharge3]
+
+    for i in range(3,4):
```

Comment:
You might provide a user input up above that lets someone running this code pick which cycle(s) to look at.

Something like `cycles = ( 3 )` would make a list, and they could just add whichever numbers they want.

Then here you would write:

```suggestion
    for i in cycles:
```

Reply:
I can't get this edit to run. Its saying "TypeError: 'tuple' object cannot be interpreted as an integer"